File: challenge-1/src/main.ts
function decodeMessage(message: string) {
  const words: string[] = message.toLowerCase().split(' ');
  const wordCount: Map<string, number> = new Map();
  const result: string[] = [];

  for (const word of words) {
    if (!wordCount.has(word)) {
      wordCount.set(word, 1);
    } else {
      wordCount.set(word, (wordCount.get(word) || 0) + 1);
    }
  }

  for (const word of words) {
    const count = wordCount.get(word);

    if (!wordCount.has(word)) {
      result.push(`${word}${count}`);
      wordCount.delete(word);
    }
  }

  return result.join('');
}

File: challenge-1/src/main.py
def decode_message(message):
    words = message.lower().split()
    
    word_count = {}
    result = []

    for word in words:
        word_key = word.lower()

        if word_key not in word_count:
            word_count[word_key] = 1
        else:
            word_count[word_key] += 1

    for word in words:
        word_key = word.lower()
        if word_key in word_count:
            result.append(word_key + str(word_count[word_key]))
            del word_count[word_key]

    decoded_message = ''.join(result)
    
    return decoded_message

with open('../data/Message_01.txt', 'r') as file:
    message = file.read()

result = decode_message(message=message)

print(result)
